feat : login page 추가

inhyuk15 opened the pull request:
# Describe your changes

Colors 변수들추가
커스텀훅 useInput추가
SignIn.tsx 추가
SignIn.Style.ts추가
Global.style.ts의 ROOT_FONT_SIZE 수정

- [x] #1 

## Checklist

- [x] 🤔 Did you follow the style guide for this project?
- [x] 🤔 Did you confirm that there was no problem with the code before PR?
- [x] 🤔 Did you comment where you needed it?

## Next Step Todo (optional)
1. 아이디 비밀번호 정보를 서버에 전달해서 관리자 검증 필요
2. 사용자 로그인 유지 여부를 세션으로 관리 필요
3. 리덕스 툴킷으로 로그인 정보 유지 필요



## Questions


- 💬 questions!
- 🤷‍♂️ check please!

inhyuk15 committed:
feat : login page 추가

okinawaa commented on src/pages/SignIn/SignIn.tsx:
```diff
@@ -1,17 +1,45 @@
 import { logo } from "@images/common";
-import { Link } from "react-router-dom";
-import { Title1 } from "~/constants/Typography";
+// import { Link } from "react-router-dom";
```
사용하지 않는 Import구문은 제거해주는 편이 좋습니다!

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+`;
+
+export const InputBox = styled.div`
+  margin-top: 0.75em;
```
Size.ts에 정의된 pxToRem을 사용하면 px수치를 입력하여 rem으로 값을 부여할 수 있습니다.
프로젝트 전반적으로 통일감있는 반응형 지원을 위해서 꼭 사용해주세요😎

inhyuk15 replied:
pxToRem을 사용해보려고 했는데 피그마로 작성된 와이어프레임에 30px로 되어있어서 pxToRem(30)을 입력했더니 엄청 큰 글자가 나와서 일단 rem을 직접 입력했습니다. pxToRem을 잘못사용한걸까요

okinawaa replied:
1920 기준으로 작성된 와이어프레임에 30px로 명시되어있어서
`pxToRem(30)` 을 작성하면 1920 모니터에서 30px로 보이게 됩니다.

1000 width인 모니터에서 본다면 (30 * 1000 / 1920)px 의 크기로 보이게 됩니다.
혹시 코드 수정 후 스크린샷 촬영해주실 수 있으신가요?

inhyuk15 replied:
<img width="1078" alt="스크린샷 2022-11-04 오전 12 41 28" src="https://user-images.githubusercontent.com/78422003/199766844-f06af1cb-3a15-4160-bc3c-374300ee3adc.png">
<img width="975" alt="스크린샷 2022-11-04 오전 12 42 14" src="https://user-images.githubusercontent.com/78422003/199767132-e150d19d-ad8d-4eee-9e5d-ded979285e76.png">



SignIn.style.ts에 SignInLabel의 수정후 수정전 비교 스크린샷입니다.

okinawaa replied:
승우님 죄송합니다. 제가 프로젝트 셋팅에 오류가 있어서 해당 이슈가 발생했었네요.
global.style.ts에서 다음과같이 *2를 삭제해주실 수 있으신가요?


![image](https://user-images.githubusercontent.com/69495129/199774039-c548138d-eef0-4e33-8f4e-da221c676211.png)

inhyuk15 replied:
넵 변경하였습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+
+export const SignInLabel = styled.text`
+  grid-row: 1;
+  font-size: 0.75rem;
```
Size.ts에 정의된 pxToRem을 사용하면 px수치를 입력하여 rem으로 값을 부여할 수 있습니다.
프로젝트 전반적으로 통일감있는 반응형 지원을 위해서 꼭 사용해주세요😎

inhyuk15 replied:
넵 이부분 수정하였습니다. 이런 폰트나 크기같은건
https://www.figma.com/file/X2XLOveZqOGfWoXawGoHYg/Untitled?node-id=0%3A1
와이어프레임에 적당히 픽셀로 적혀있는데 margin같은건 적혀있지 않아서 0.75rem 같은건 그대로 놔뒀습니다.

okinawaa replied:
<img width="942" alt="image" src="https://user-images.githubusercontent.com/69495129/200194704-7bb22fb6-bf68-4dc6-86f5-826aef8af6ee.png">

혹시 이부분이 폰트사이즈 아닌가요?!

okinawaa replied:
<img width="1223" alt="image" src="https://user-images.githubusercontent.com/69495129/200194832-c5a35207-f132-4992-8e52-476568e0603b.png">

현재 1920px 사이즈에서 
0.75rem이면 10*0.75 = 7.5px 으로 렌더링되고있어서 사용자 경험이 좋지 않습니다.

inhyuk15 replied:
1rem = 16px 기준으로 0.75rem을 pxToRem(12)로 변경하였습니다.

okinawaa replied:
좋습니다~

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+  margin-top: 0.75em;
+`;
+
+export const SignInLabel = styled.text`
```
html 태그중 text태그는 존재하지 않는 것으로 알고 있습니다.
개발자 도구에서 하기와 같은 오류가 발생하고 있습니다. 일반적으로 텍스트를 사용할 경우에는 
Typography에 정의된 값을 이용해주시면 감사하겠습니다🤩

<img width="987" alt="image" src="https://user-images.githubusercontent.com/69495129/199744752-ffeabc2c-7061-4206-98c3-e1a595029094.png">

inhyuk15 replied:
넵 이 부분 수정하겠습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+`;
+
+export const SignInLabel = styled.text`
+  grid-row: 1;
```
해당 속성을 개발자도구에서 제거해도 아무런 영향이 없습니다. 
혹시 어떠한 행동을 하는 코드인지 말씀해주시면 감사하겠습니다

inhyuk15 replied:
input태그 위의 아이디, 비밀번호 레이블을 만들때 사용합니다. 이게 없으면 input tag만 있게 될거에요

okinawaa replied:
어떠한 말씀이신지 조금만 더 설명 가능할까요?

<img width="1216" alt="image" src="https://user-images.githubusercontent.com/69495129/200194807-f75fc08f-0b7c-4284-97d2-c455da8351f3.png">


제가 지워봤을땐 아무런 변화가없어서요!

inhyuk15 replied:
지금 지웠을 때 글씨 크기가 많이 작아진 것으로 보이는데 
<img width="207" alt="스크린샷 2022-11-07 오전 8 15 41" src="https://user-images.githubusercontent.com/78422003/200200791-6072c6fb-4b1e-425c-86c5-62d5139f6a21.png">
여기 입력받는 값 위에 Label, 그러니까 아이디 라고 적힌 부분이 그것입니다.
<img width="317" alt="스크린샷 2022-11-07 오전 8 15 18" src="https://user-images.githubusercontent.com/78422003/200200836-fc9a3c8c-4f71-42a0-acee-d8d02f582f2d.png">
<img width="509" alt="스크린샷 2022-11-07 오전 8 15 12" src="https://user-images.githubusercontent.com/78422003/200200843-3feececa-71cf-4fbc-9d52-b78b7ac42913.png">

색깔이랑 크기가 필요해서 단독 텍스트만 쓰지 않고 SignInLabel 컴포넌트로 묶습니다.

okinawaa replied:
승우님 리뷰가 달리면 바로 위에 코드에 대해서 리뷰를 남긴것입니다.
![image](https://user-images.githubusercontent.com/69495129/200202830-56cfbcbc-19b8-41c7-b949-cede1619945f.png)


저는 grid-row : 1 이 필요한지에 대해서 궁금한것입니다!

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+export const SignInLabel = styled.text`
+  grid-row: 1;
+  font-size: 0.75rem;
+  font-weight: 600;
```
일관성있는 프로젝트 구성을 위해 [magic number](https://drehzr.tistory.com/1136) 작성은 최대한 지양하는것이 좋습니다.
Variables에 `BOLD_WEIGHT` `EXTRA_BOLD_WEIGHT` 정의해두었으니 사용하시면 될 것 같습니다.

inhyuk15 replied:
넵 이거로 사용하겠습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+  grid-row: 1;
+  font-size: 0.75rem;
+  font-weight: 600;
+  color: #838383;
```
색상 해쉬값을 직접 사용하는것은 추후 프로젝트 유지보수에 좋지 않습니다.
Colors.ts 에 해당 값을 정의한 뒤 사용하시면 일관성있게 코드를 구성할 수 있습니다.

저라면 이 값을

Colors.ts 에
gray83: "#838383"

으로 정의한 뒤 불러와서 Colors.gray83으로 사용할 것 같습니다!

inhyuk15 replied:
넵 이 부분 수정했습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+`;
+
+export const SignInInput = styled.input`
+  border: 1px solid #afafaf;
```
색상 해쉬값을 직접 사용하는것은 추후 프로젝트 유지보수에 좋지 않습니다.
Colors.ts 에 해당 값을 정의한 뒤 사용하시면 일관성있게 코드를 구성할 수 있습니다.

저라면 이 값을

Colors.ts 에
grayAF: "#AFAFAF"

으로 정의한 뒤 불러와서 Colors.gray83으로 사용할 것 같습니다!

inhyuk15 replied:
넵 이것으로 수정햇습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+
+export const SignInInput = styled.input`
+  border: 1px solid #afafaf;
+  border-radius: 5px;
```
Size.ts에 정의된 pxToRem을 사용하면 px수치를 입력하여 rem으로 값을 부여할 수 있습니다.
프로젝트 전반적으로 통일감있는 반응형 지원을 위해서 꼭 사용해주세요😎

inhyuk15 replied:
넵 이 부분 수정하였습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+  border: 1px solid #afafaf;
+  border-radius: 5px;
+  padding: 0 65px 0 5px;
+  margin-top: 0.2rem;
```
Size.ts에 정의된 pxToRem을 사용하면 px수치를 입력하여 rem으로 값을 부여할 수 있습니다.
프로젝트 전반적으로 통일감있는 반응형 지원을 위해서 꼭 사용해주세요😎

inhyuk15 replied:
이 부분 수정하였습니다. 0은 그대로 냅두고 65px만 pxToRem(65) 이런식으로 수정하였습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+export const SignInInput = styled.input`
+  border: 1px solid #afafaf;
+  border-radius: 5px;
+  padding: 0 65px 0 5px;
```
Size.ts에 정의된 pxToRem을 사용하면 px수치를 입력하여 rem으로 값을 부여할 수 있습니다.
프로젝트 전반적으로 통일감있는 반응형 지원을 위해서 꼭 사용해주세요😎

inhyuk15 replied:
이 부분 수정하였습니다. 0은 그대로 냅두고 65px만 pxToRem(65) 이런식으로 수정하였습니다.

okinawaa replied:
좋습니다 0 은 그대로 두는게 맞습니다!

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+  height: 35px;
+  outline: none;
+  ::placeholder {
+    color: #c3c3c3;
```
이 색상값도 Colors.ts 에 정의후 사용해주시면 감사하겠습니다.

inhyuk15 replied:
넵 수정해서 다시 pr드리겟습니다

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+  margin-top: 0.2rem;
+  width: 100%;
+  height: 35px;
+  outline: none;
```
기본적으로 Global.styles.ts에서  input태그는 Default로 Outline을 None으로 지정해주었는데
여기서 한번 더 지정해주신 이유가 있으신가요?

inhyuk15 replied:
아하 몰랐었네요 이 부분도 수정하겠습니다

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -11,3 +11,45 @@ export const Container = styled(FlexCenter)`
+`;
+
+export const LoginButton = styled.button`
+  top: 0;
+  right: 0;
+  border: none;
+  border-radius: 5px;
+  width: 100%;
+  height: 35px;
+  margin-top: 0.75rem;
+  background-color: #001e75;
+  color: #fff;
+  font-size: 0.75rem;
```
이 부분도 위와 같은 컨벤션으로 사용해주세요!

inhyuk15 replied:
color를 Color.ts에 있는값으로 고치겠습니다

okinawaa commented on src/pages/SignIn/SignIn.tsx:
```diff
@@ -1,17 +1,45 @@
 import * as Styled from "./SignIn.style";
 
 const SignIn = () => {
+  const [adminId, setAdminId] = useState("");
+  const [adminPw, setAdminPw] = useState("");
```
input의 값을 컨트롤 할때 주로 
useInput이라는 커스텀훅을 만들어서 hooks folder안에 저장한 뒤 사용하는것이 관례입니다.
프로젝트를 진행하면서 input값을 컨트롤해야하는경우는 수도 없이 많고 관리자 페이지는 더더욱이 그렇기 때문입니다.

[useInput](https://velog.io/@choidy180/React-Hook-1-useInput)은 리액트의 대표 커스텀 훅인 만큼 검색해봐도 수많은 예제가 있으므로 그것으로 대체하여 상태를 관리해주시면 더 좋을 것 같습니다.

inhyuk15 replied:
커스텀훅으로 수정하겠습니다. 그러면 커스텀훅에 대한 ts파일을 따로 만들게 되겟네요. 그렇게 해도 될까요

okinawaa replied:
네 !! 너무 좋습니다.

![image](https://user-images.githubusercontent.com/69495129/199774789-ae121862-fe0b-4bac-b2a7-2f23ab459990.png)


다음과 같은 폴더구조를 따라주시면됩니다

inhyuk15 replied:
커스텀훅으로 수정하였습니다. form에서 제출시에 초기화하는 것까지 넣어서 커스텀훅으로 만들었고
[adminId, setAdminId] 이렇게 되어있었던 것을 어짜피 Input받는 것이 더 중요하다고 생각해서
idInput 이런식으로 수정하였습니다.

okinawaa commented on src/pages/SignIn/SignIn.tsx:
```diff
@@ -1,17 +1,45 @@
+            }}
+          />
+        </Styled.InputBox>
+        <Styled.InputBox>
+          <Styled.SignInLabel> 비밀번호 </Styled.SignInLabel>
+          <Styled.SignInInput
+            type="text"
+            placeholder="비밀번호"
+            onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+              setAdminPw(e.target.value);
+            }}
+          />
```
비밀번호는 masking된 형태로 사용자가 확인 할 수 있도록 해주세요.
사용자가 보안에 위험할 수 있습니다.

inhyuk15 replied:
넵 안보이게끔 수정하겠습니다.

okinawaa reviewed (changes requested):
1차 리뷰 남겼습니다.
긴 작업하시느라 고생 많으셨습니다😎

inhyuk15 committed:
Fix : Colors.ts

자주 사용하는 색깔들 추가했습니다.
gray83, grayC3, grayAF, southenBlue

---
Fix: Global.style.ts

html의 font-size의 *2를 제거하였습니다.

---
Fix: SiginIn.style.ts

px 숫자 그대로 되있는 것을 ~/constants/Size의 pxToRem을 이용하여 바꾸었습니다.

---
Add: ~/hooks/useInputs.ts

SignIn.tsx에서 사용하는 유저 아이디 비밀번호를 받기 위한 훅을 따로 만들었습니다.

---
Fix: SignIn.tsx

커스텀 훅을 이용하여 입력받게 변경하였고 Form에 전달후 빈값으로 바꾸게 수정하였습니다.

okinawaa commented on src/hooks/useInput.ts:
```diff
@@ -0,0 +1,27 @@
+import { useCallback, useState } from "react";
+
+interface IuseInputProps {
+  type: string;
+  defValue: string | number;
+}
```
이 파일 내에서 Props로 받는것은 useInput의 Props로 들어가는것이 너무나 명확하기 때문에 
prefix가 필요없이 Props로만 인터페이스 명을 정해도 useInput에 들어가는 Props라는 사실은 알 수 있을 것 같습니다

inhyuk15 replied:
넵 인터페이스 이름 수정했습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -1,55 +1,59 @@
 `;
-
-export const SignInLabel = styled.text`
+// font-size: ${pxToRem(30}; global.style.ts 수정후에도 적용이 되지 않습니다.
```
불필요한 주석은 삭제 부탁드리겠습니다!

inhyuk15 replied:
넵 해당주석 삭제하였습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -1,55 +1,59 @@
 export const LoginButton = styled.button`
   top: 0;
   right: 0;
   border: none;
```
이 세개가 필요한 속성인가요?

inhyuk15 replied:
수정했습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -1,55 +1,59 @@
   width: 100%;
-  height: 35px;
+  height: ${pxToRem(35)};
   margin-top: 0.75rem;
```
pxToRem으로 수정부탁드리겠습니다.

inhyuk15 replied:
넵 해당 파일에 px단독으로 쓰이는 건 이제 없습니다.

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -1,55 +1,59 @@
   margin-top: 0.75rem;
-  background-color: #001e75;
+  background-color: ${Colors.SouthernBlue};
   color: #fff;
```
Colors.white 사용부탁드리겠습니다

inhyuk15 replied:
넵 수정하였습니다.

okinawaa commented on src/pages/SignIn/SignIn.tsx:
```diff
@@ -1,45 +1,37 @@
-  const [adminId, setAdminId] = useState("");
-  const [adminPw, setAdminPw] = useState("");
+  const { inputProps: idInput, resetValue: resetIdInput } = useInput({
+    type: "text",
```
type도 넣는것 좋네요!

inhyuk15 replied:
:D

okinawaa reviewed (changes requested):
수고하셨습니다~!

inhyuk15 committed:
Fix: useInput.ts

IuseInputProps 이름을 IProps로 변경.

---
Fix: SignIn.style.ts

px과 rem을 사용한 것들 모두 pxToRem()을 이용하여 수정.
54번째 줄 매직넘버를 사용하던 코드를 Color.white로 변경.
필요없는 주석 제거.
LoginButton의 의미없는 속성 제거.

okinawaa commented on src/hooks/useInput.ts:
```diff
@@ -1,11 +1,11 @@
 import { useCallback, useState } from "react";
 
-interface IuseInputProps {
+interface IProps {
```
I라는 prefix도 필요없을 것 같은데 
특별한 이유가 있으신가요?
이미 앞에 `interface` keyword가 작성되어있어서 interface라는 사실은 누구나 잘 알 수 있을 것 같습니다!

inhyuk15 replied:
타입스크립트에선 이런 간단한 거에 대해서는 I를 쓰지 않나요. 잘몰라서 일단 붙였습니다.
일반적으로는 인터페이스앞에 I를 붙여야하는걸로 알고 있습니다.

okinawaa replied:
그건 아마 프로젝트 컨벤션에 따라서 달라질것 같은데 안붙혀도 잘 동작은합니다!
제 생각엔 위에 말씀드린것처럼 
>이미 앞에 interface keyword가 작성되어있어서 interface라는 사실은 누구나 잘 알 수 있을 것 같습니다!

라고 생각하는데
승우님 생각은 어떠신가요?!

inhyuk15 replied:
type 네이밍 규칙과 인터페이스 네이밍 규칙을 정해주셔야할 것 같습니다.
일단은 저는 인터페이스 네이밍에 I를 안붙이는 경우를 못봐서 I를 붙여야한다고 생각합니다. 일반 컴포넌트와 차별성이 전혀없기 때문에요.
ps. 저는 승우님이 아니라 inhyuk입니당

okinawaa replied:
ㅋㅋㅋㅋㅋㅋㅋㅋ ㅠㅠ 죄송해요 승우님리뷰를 많이 달았다보니깐 인혁님 ㅠㅠㅠ
컴포넌트가 사용되는 부분에 인터페이스가 사용될 수도 없고 Props라는것 자체로 인터페이스이다! 라는것을 알 수 있기때문에 안붙히는걸로 컨벤션을 정하겠습니다! 인혁님 괜찮으신가요?!

okinawaa replied:
괜찮으시다면 수정 부탁드리겠습니다!

inhyuk15 replied:
수정했습니다!

inhyuk15 committed:
Fix : useInput.ts

IProps => Props 인터페이스 이름 변경

okinawaa commented on src/pages/SignIn/SignIn.style.ts:
```diff
@@ -1,13 +1,56 @@
+
+export const SignInLabel = styled(Text)`
+  grid-row: 1;
+  font-size: ${pxToRem(30)};
+  font-weight: ${EXTRA_BOLD_WEIGHT};
+  color: ${Colors.gray83};
```
Text에는 기본적으로 
fontSize, weight, _color 
props가 있습니다. 그것을 사용해주세요!

inhyuk15 replied:
해창님이 주신 와이어프레임에 있는 피그마에서 30px로 되어있는데 
<img width="317" alt="스크린샷 2022-11-07 오전 8 15 18" src="https://user-images.githubusercontent.com/78422003/200225496-aa349085-0af2-455e-8f60-0507cc74eefa.png">
<img width="352" alt="스크린샷 2022-11-07 오후 1 16 27" src="https://user-images.githubusercontent.com/78422003/200225501-9e07d556-b2f2-40b8-b6fa-efc12b06b1ed.png">
Typography.ts에 30px가 없어서 임의로 SignIn.style.ts에 만들어서 사용하고 있습니다. 현재 SignIn.style.ts에 있는 내용을 통째로 Typography.ts에 옮겨서 Typography에서 가져와서 이용할 수도 있지만
SignIn.style.ts에 SignInLabel값을 남겨둘것인지 옮길것인지 고민되네요.
어떻게 생각하시나요?

일단 SignIn 관련파일말고 다른 파일에 대해서는 Typography.ts에 있는 객체들을 적극이용하겠습니다.

okinawaa replied:
![image](https://user-images.githubusercontent.com/69495129/200226960-57660d25-6d90-475c-8ae6-4e476599f32f.png)

제가 언급드린 부분은 이거였습니다.

사용하는 부분에서 
<Text size={30} color={Colors.gray83} weight={EXTRA_BOLD_WEIGHT/> 으로 사용하면 이 스타일 컴포넌트를 선언할 필요도 없습니다.

okinawaa replied:
![image](https://user-images.githubusercontent.com/69495129/200227081-a9b9759a-f2fa-4aa8-8576-b943170aed3d.png)


grid-row도 전부터 말씀드렸듯이 UI에 아무런 영향을 주지 않습니다.

inhyuk15 replied:
이걸 이용하게 되면 SignIn.tsx에 직접 값을 넣게 되지 않을까요
style과 tsx의 내용이 뒤섞일 거 같아서 염려가 됩니다.
아니면 제가 사용법을 잘못 알고 있을까요 ts문법을 잘 알지 못해서 그렇게 생각을 못했습니다.
예를 들어
Color.gray83을 써야는데
SignIn.tsx에 
SignInLabel color={Colors.gray83} 아이디 /SignInLabel
이런식으로 작성하게 되지 않을까요

okinawaa replied:
맞아요 인혁님이 정확히 알고계십니다..
저도 항상 style 적 요소가 tsx파일에 있다는것에 의문을 가졌습니다. 
그래서 styled-component 가 아닌 또 다른 CSS-in-js 인 emotion에서는 아예 inline styling을 추구하는 이유중 하나라고 생각합니다.
항상 개발하다보면 어쩔 수 없이 필수불가결하게  tsx에 스타일적 코드가 많이 생기게 되는데  이 부분은 인혁님이 보여주신 코드 그대로 맞습니다! 
인혁님께서 그게 싫고 확실하게 분리하고 싶으시다면 

```tsx
export const SignInLabel = styled(Text).attrs({
  size: 30,
  weight: EXTRA_BOLD_WEIGHT,
  color: Colors.gray83
})``;
```

이렇게 선언하셔서 사용하셔도 됩니다. 하지만 저는 tsx에 스타일적 코드가 섞여도 괜찮다고 생각합니다. 어쩔 수 없이 마크업과 스타일링은 공존해야하기때문이고 프로젝트가 커질수록 자바스크립트로직으로 스타일을 제어해야할 경우가 많이 생기기때문입니다!

inhyuk15 replied:
확인했습니다. 올려주신 코드대로 기존 Text를 잘 상속하게끔 변경하였습니다.
```typescript:SignIn.style.ts
export const SignInLabel = styled(Text).attrs({
  size: pxToRem(30),
  weight: EXTRA_BOLD_WEIGHT,
  color: Colors.gray83
})``;
```

inhyuk15 committed:
Fix: SignIn.style.ts

SignInLabel를 Typography의 Text를 상속해 필요한 값만 바꾸게 수정

inhyuk15 committed:
Fix: SignIn.style.ts

SignInInput의 의미없는 속성 제거
